src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -22,7 +25,16 @@ public GrainCreator(IGrainRuntime grainRuntime, IServiceProvider services = null
         {
             _grainRuntime = grainRuntime;
             _services = services;
```

Comment:
what's the goal for that? you mean using IGrainRuntime.ServiceProvider instead?
Sure, although some optimizations here rely on the service provider never changing under our feet